internal/dot/graph.go

```diff
@@ -20,53 +20,166 @@
 
 package dot
 
-import "fmt"
+import (
+	"fmt"
+	"reflect"
+)
 
 // Ctor encodes a constructor provided to the container for the DOT graph.
 type Ctor struct {
-	Name    string
-	Package string
-	File    string
-	Line    int
-	Params  []*Node
-	Results []*Node
+	Name       string
+	Package    string
+	File       string
+	Line       int
+	Params     []*Param
+	GroupParam []*Group
+	Results    []*Result
+}
+
+// Param is a parameter node in the graph.
+type Param struct {
+	*Node
+	Optional bool
+}
+
+// Result is a result node in the graph.
+type Result struct {
+	*Node
+	GroupIndex int
```

Comment:
nit: add a comment here explaining the purpose of GroupIndex because it won't
make much sense to someone who wasn't part of that discussion.

Reply:
sure, I realized it's not really self-explanatory. I added comments explaining GroupIndex.